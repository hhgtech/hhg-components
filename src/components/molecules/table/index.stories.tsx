import React from 'react'

import { Story } from '@storybook/react/types-6-0'

import * as Table from '../table'

import MockTheme from './../../../mockTheme'

export default {
  component: Table,
  title: 'Molecules/Table',
}

const Template: Story = (args) => (
  <Table.Table {...args}>
    <Table.Head isSticky>
      <Table.Cell isTableHead>Header Cell one</Table.Cell>
      <Table.Cell isTableHead>Header Cell Two</Table.Cell>
      <Table.Cell isTableHead>Header Cell Tree</Table.Cell>
      <Table.Cell isTableHead>Header Cell Four</Table.Cell>
      <Table.Cell isTableHead>Header Cell Five</Table.Cell>
    </Table.Head>
    <Table.Body isStriped isHoverable>
      <Table.Row>
        <Table.Cell>Cell one</Table.Cell>
        <Table.Cell>Cell Two</Table.Cell>
        <Table.Cell>Cell Three</Table.Cell>
        <Table.Cell>Cell Four</Table.Cell>
        <Table.Cell>Cell Dive</Table.Cell>
      </Table.Row>
      <Table.Row>
        <Table.Cell>Cell one</Table.Cell>
        <Table.Cell>Cell Two</Table.Cell>
        <Table.Cell>Cell Three</Table.Cell>
        <Table.Cell>Cell Four</Table.Cell>
        <Table.Cell>Cell Dive</Table.Cell>
      </Table.Row>
      <Table.Row>
        <Table.Cell>Cell one</Table.Cell>
        <Table.Cell>Cell Two</Table.Cell>
        <Table.Cell>Cell Three</Table.Cell>
        <Table.Cell>Cell Four</Table.Cell>
        <Table.Cell>Cell Dive</Table.Cell>
      </Table.Row>
      <Table.Row>
        <Table.Cell>Cell one</Table.Cell>
        <Table.Cell>Cell Two</Table.Cell>
        <Table.Cell>Cell Three</Table.Cell>
        <Table.Cell>Cell Four</Table.Cell>
        <Table.Cell>Cell Dive</Table.Cell>
      </Table.Row>
      <Table.Row>
        <Table.Cell>Cell one</Table.Cell>
        <Table.Cell>Cell Two</Table.Cell>
        <Table.Cell>Cell Three</Table.Cell>
        <Table.Cell>Cell Four</Table.Cell>
        <Table.Cell>Cell Dive</Table.Cell>
      </Table.Row>
      <Table.Row>
        <Table.Cell>Cell one</Table.Cell>
        <Table.Cell>Cell Two</Table.Cell>
        <Table.Cell>Cell Three</Table.Cell>
        <Table.Cell>Cell Four</Table.Cell>
        <Table.Cell>Cell Dive</Table.Cell>
      </Table.Row>
      <Table.Row>
        <Table.Cell>Cell one</Table.Cell>
        <Table.Cell>Cell Two</Table.Cell>
        <Table.Cell>Cell Three</Table.Cell>
        <Table.Cell>Cell Four</Table.Cell>
        <Table.Cell>Cell Dive</Table.Cell>
      </Table.Row>
      <Table.Row>
        <Table.Cell>Cell one</Table.Cell>
        <Table.Cell>Cell Two</Table.Cell>
        <Table.Cell>Cell Three</Table.Cell>
        <Table.Cell>Cell Four</Table.Cell>
        <Table.Cell>Cell Dive</Table.Cell>
      </Table.Row>
      <Table.Row>
        <Table.Cell>Cell one</Table.Cell>
        <Table.Cell>Cell Two</Table.Cell>
        <Table.Cell>Cell Three</Table.Cell>
        <Table.Cell>Cell Four</Table.Cell>
        <Table.Cell>Cell Dive</Table.Cell>
      </Table.Row>
      <Table.Row>
        <Table.Cell>Cell one</Table.Cell>
        <Table.Cell>Cell Two</Table.Cell>
        <Table.Cell>Cell Three</Table.Cell>
        <Table.Cell>Cell Four</Table.Cell>
        <Table.Cell>Cell Dive</Table.Cell>
      </Table.Row>
      <Table.Row>
        <Table.Cell>Cell one</Table.Cell>
        <Table.Cell>Cell Two</Table.Cell>
        <Table.Cell>Cell Three</Table.Cell>
        <Table.Cell>Cell Four</Table.Cell>
        <Table.Cell>Cell Dive</Table.Cell>
      </Table.Row>
      <Table.Row>
        <Table.Cell>Cell one</Table.Cell>
        <Table.Cell>Cell Two</Table.Cell>
        <Table.Cell>Cell Three</Table.Cell>
        <Table.Cell>Cell Four</Table.Cell>
        <Table.Cell>Cell Dive</Table.Cell>
      </Table.Row>
      <Table.Row>
        <Table.Cell>Cell one</Table.Cell>
        <Table.Cell>Cell Two</Table.Cell>
        <Table.Cell>Cell Three</Table.Cell>
        <Table.Cell>Cell Four</Table.Cell>
        <Table.Cell>Cell Dive</Table.Cell>
      </Table.Row>
      <Table.Row>
        <Table.Cell>Cell one</Table.Cell>
        <Table.Cell>Cell Two</Table.Cell>
        <Table.Cell>Cell Three</Table.Cell>
        <Table.Cell>Cell Four</Table.Cell>
        <Table.Cell>Cell Dive</Table.Cell>
      </Table.Row>
      <Table.Row>
        <Table.Cell>Cell one</Table.Cell>
        <Table.Cell>Cell Two</Table.Cell>
        <Table.Cell>Cell Three</Table.Cell>
        <Table.Cell>Cell Four</Table.Cell>
        <Table.Cell>Cell Dive</Table.Cell>
      </Table.Row>
      <Table.Row>
        <Table.Cell>Cell one</Table.Cell>
        <Table.Cell>Cell Two</Table.Cell>
        <Table.Cell>Cell Three</Table.Cell>
        <Table.Cell>Cell Four</Table.Cell>
        <Table.Cell>Cell Dive</Table.Cell>
      </Table.Row>
      <Table.Row>
        <Table.Cell>Cell one</Table.Cell>
        <Table.Cell>Cell Two</Table.Cell>
        <Table.Cell>Cell Three</Table.Cell>
        <Table.Cell>Cell Four</Table.Cell>
        <Table.Cell>Cell Dive</Table.Cell>
      </Table.Row>
    </Table.Body>
  </Table.Table>
)

export const FullTable = Template.bind({})

FullTable.args = {
  horizontalPadding: '19px',
  verticalPadding: '30px',
}

const Header: Story = (args) => (
  <Table.Table>
    <Table.Head>
      <th>Header Cell one</th>
      <th>Header Cell Two</th>
      <th>Header Cell Three</th>
      <th>Header Cell Four</th>
      <th>Header Cell Five</th>
    </Table.Head>
  </Table.Table>
)

export const TableHeader = Header.bind({})

TableHeader.args = {
  isSticky: true,
}

const Row: Story = (args) => (
  <Table.Table>
    <Table.Row {...args}>
      <Table.Cell>Cell one</Table.Cell>
      <Table.Cell>Cell Two</Table.Cell>
      <Table.Cell>Cell Three</Table.Cell>
      <Table.Cell>Cell Four</Table.Cell>
      <Table.Cell>Cell Dive</Table.Cell>
    </Table.Row>
    <Table.Row>
      <Table.Cell>Cell one</Table.Cell>
      <Table.Cell>Cell Two</Table.Cell>
      <Table.Cell>Cell Three</Table.Cell>
      <Table.Cell>Cell Four</Table.Cell>
      <Table.Cell>Cell Dive</Table.Cell>
    </Table.Row>
    <Table.Row>
      <Table.Cell>Cell one</Table.Cell>
      <Table.Cell>Cell Two</Table.Cell>
      <Table.Cell>Cell Three</Table.Cell>
      <Table.Cell>Cell Four</Table.Cell>
      <Table.Cell>Cell Dive</Table.Cell>
    </Table.Row>
    <Table.Row>
      <Table.Cell>Cell one</Table.Cell>
      <Table.Cell>Cell Two</Table.Cell>
      <Table.Cell>Cell Three</Table.Cell>
      <Table.Cell>Cell Four</Table.Cell>
      <Table.Cell>Cell Dive</Table.Cell>
    </Table.Row>
  </Table.Table>
)

export const TableRow = Row.bind({})

TableRow.args = {
  isActive: true,
}

const Body: Story = (args) => (
  <Table.Table>
    <Table.Body {...args}>
      <Table.Row>
        <Table.Cell>Cell one</Table.Cell>
        <Table.Cell>Cell Two</Table.Cell>
        <Table.Cell>Cell Three</Table.Cell>
        <Table.Cell>Cell Four</Table.Cell>
        <Table.Cell>Cell Dive</Table.Cell>
      </Table.Row>
      <Table.Row>
        <Table.Cell>Cell one</Table.Cell>
        <Table.Cell>Cell Two</Table.Cell>
        <Table.Cell>Cell Three</Table.Cell>
        <Table.Cell>Cell Four</Table.Cell>
        <Table.Cell>Cell Dive</Table.Cell>
      </Table.Row>
      <Table.Row>
        <Table.Cell>Cell one</Table.Cell>
        <Table.Cell>Cell Two</Table.Cell>
        <Table.Cell>Cell Three</Table.Cell>
        <Table.Cell>Cell Four</Table.Cell>
        <Table.Cell>Cell Dive</Table.Cell>
      </Table.Row>
      <Table.Row>
        <Table.Cell>Cell one</Table.Cell>
        <Table.Cell>Cell Two</Table.Cell>
        <Table.Cell>Cell Three</Table.Cell>
        <Table.Cell>Cell Four</Table.Cell>
        <Table.Cell>Cell Dive</Table.Cell>
      </Table.Row>
    </Table.Body>
  </Table.Table>
)

export const TableBody = Body.bind({})

TableBody.args = {
  isStriped: true,
  isHoverable: false,
}

const Cell: Story = (args) => (
  <Table.Table>
    <Table.Row>
      <Table.Cell {...args}>Header Cell one</Table.Cell>
    </Table.Row>
  </Table.Table>
)

export const TableCell = Cell.bind({})

TableCell.args = {
  textAlign: 'right',
  isTableHead: false,
}
