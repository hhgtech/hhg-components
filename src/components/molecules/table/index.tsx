export { Table } from './table'
export { Cell } from './tableCell'
export { Head } from './tableHead'
export { Row } from './tableRow'
export { Body } from './tableBody'
