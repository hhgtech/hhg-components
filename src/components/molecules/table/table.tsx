import React, { CSSProperties, ReactNode } from 'react'

import styled from 'styled-components'

type Props = {
  children: ReactNode
  horizontalPadding?: string
  verticalPadding?: string
  className?: string
  style?: CSSProperties
}

const StyledTable = styled.table`
  --horizontal-spacing: ${(props) => props.theme.horizontalPadding || '19px'};
  --vertical-spacing: ${(props) => props.theme.verticalPadding || '26px'};
  width: 100%;
  position: relative;
`

const Table = ({
  children,
  className,
  style,
  horizontalPadding,
  verticalPadding,
}: Props) => {
  return (
    <StyledTable
      className={className}
      style={style}
      theme={{ horizontalPadding, verticalPadding }}
    >
      {children}
    </StyledTable>
  )
}

export { Table }
