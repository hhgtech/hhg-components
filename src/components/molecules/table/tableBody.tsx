import React, { CSSProperties, ReactNode } from 'react'
import styled from 'styled-components'
import { ThemeProps, ThemeType } from 'theme'

type Props = {
  children: ReactNode
  isHoverable?: boolean
  isStriped?: boolean
  style?: CSSProperties
  className?: string
}

const StyledTableBody = styled.tbody`
  position: relative;
  &[data-is-striped] {
    > tr:nth-child(odd) {
      background-color: ${(props) => props.theme.colors.neutral50};
    }
  }
  &[data-is-hoverable] {
    > tr:hover {
      background-color: ${(props) => props.theme.colors.neutral100};
    }
  }
`

const Body = ({
  children,
  className,
  style,
  isHoverable,
  isStriped,
}: Props) => {
  return (
    <StyledTableBody
      className={className}
      style={style}
      data-is-striped={isStriped || undefined}
      data-is-hoverable={isHoverable || undefined}
    >
      {children}
    </StyledTableBody>
  )
}

export { Body }
