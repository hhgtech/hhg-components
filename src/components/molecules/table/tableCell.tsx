import React, { CSSProperties, ReactNode } from 'react'
import styled from 'styled-components'

type Props = {
  children: ReactNode
  textAlign?: 'left' | 'center' | 'right'
  className?: string
  isTableHead?: boolean
  style?: CSSProperties
}

const StyledTableCell = styled.td`
  padding: var(--vertical-spacing) var(--horizontal-spacing);
  vertical-align: middle;

  &[data-text-aling='left'] {
    text-align: left;
  }
  &[data-text-aling='right'] {
    text-align: right;
  }
  &[data-text-aling='center'] {
    text-align: center;
  }
`

const Cell = ({
  children,
  className,
  style,
  textAlign,
  isTableHead,
}: Props) => {
  return (
    <StyledTableCell
      className={className}
      style={style}
      data-text-align={textAlign ? textAlign : 'left'}
      as={isTableHead ? 'th' : 'td'}
    >
      {children}
    </StyledTableCell>
  )
}

export { Cell }
