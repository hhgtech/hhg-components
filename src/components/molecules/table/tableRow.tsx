import React, { CSSProperties, ReactNode } from 'react'

import { ThemeProps } from 'misc/theme'
import styled from 'styled-components'

type Props = {
  children: ReactNode
  isActive?: boolean
  className?: string
  style?: CSSProperties
}

const StyledTableRow = styled.tr`
  border-bottom: 1px solid ${(props: ThemeProps) => props.theme.colors.gray200};

  &[data-is-active] {
    background-color: ${(props: ThemeProps) => props.theme.colors.gray200};
  }
`

const Row = ({ children, className, style, isActive }: Props) => {
  return (
    <StyledTableRow
      className={className}
      data-is-active={isActive || undefined}
      style={style}
    >
      {children}
    </StyledTableRow>
  )
}

export { Row }
