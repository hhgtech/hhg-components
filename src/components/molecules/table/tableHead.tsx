import React, { CSSProperties, ReactNode } from 'react'

import { ThemeProps } from 'misc/theme'
import styled from 'styled-components'

import * as Table from '../table'

type Props = {
  children: ReactNode
  className?: string
  isSticky?: boolean
  style?: CSSProperties
}

const StyledTableHead = styled.thead`
  th {
    padding: var(--vertical-spacing) var(--horizontal-spacing);
    text-align: left;

    color: ${(props: ThemeProps) => props.theme.colors.gray500};
    font-weight: ${(props: ThemeProps) => props.theme.sizes.fwBold};
  }
  &[data-is-sticky] {
    th {
      position: sticky;
      top: 0;
      background: white;
      border-bottom: 2px solid
        ${(props: ThemeProps) => props.theme.colors.gray200};
    }
  }
`

const Head = ({ isSticky, children, className, style }: Props) => {
  return (
    <StyledTableHead
      className={className}
      data-is-sticky={isSticky || undefined}
      style={style}
    >
      <Table.Row>{children}</Table.Row>
    </StyledTableHead>
  )
}

export { Head }
