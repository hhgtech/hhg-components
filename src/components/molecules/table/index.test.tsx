import React from 'react'

import { render, cleanup } from '@testing-library/react'
import 'jest-styled-components'

import * as Table from '../table'
import MockTheme from './../../../mockTheme'

afterEach(cleanup)

test('it renders', () => {
  const dropdown = render(
    <MockTheme>
      <Table.Table>
        <Table.Head isSticky>
          <th>Header one</th>
          <th>Header two</th>
          <th>Header three</th>
        </Table.Head>
      </Table.Table>
    </MockTheme>,
  )

  const self = dropdown.container.firstChild

  expect(self).toMatchSnapshot()

  expect(self.firstChild).toHaveAttribute('data-is-sticky', 'true')
})
