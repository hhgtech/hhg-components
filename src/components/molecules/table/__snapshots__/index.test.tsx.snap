// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders 1`] = `
.c0 {
  --horizontal-spacing: 19px;
  --vertical-spacing: 26px;
  width: 100%;
  position: relative;
}

.c2 {
  border-bottom: 1px solid #d9d9d9;
}

.c2[data-is-active] {
  background-color: #d9d9d9;
}

.c1 th {
  padding: var(--vertical-spacing) var(--horizontal-spacing);
  text-align: left;
  color: #737373;
  font-weight: 700;
}

.c1[data-is-sticky] th {
  position: -webkit-sticky;
  position: sticky;
  top: 0;
  background: white;
  border-bottom: 2px solid #d9d9d9;
}

<table
  class="c0"
>
  <thead
    class="c1"
    data-is-sticky="true"
  >
    <tr
      class="c2"
    >
      <th>
        Header one
      </th>
      <th>
        Header two
      </th>
      <th>
        Header three
      </th>
    </tr>
  </thead>
</table>
`;
